packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
+    super();
+  }
+
+  public get description(): string {
+    return `output\
+${this.outputName ? ' with name ' + this.outputName : ''}\
+${this.exportName ? ' with export name ' + this.exportName : ''}\
+${this.outputValue ? ' with value ' + JSON.stringify(this.outputValue) : ''}\
+`;
+  }
+
+  public assertUsing(inspector: StackInspector): boolean {
+    if (!this.outputName && !this.exportName) {
+      throw new Error('At least one of [outputName, exportName] should be defined');
+    }
+    if (!('Outputs' in inspector.value)) {
+      return false;
+    }
+    return (!this.outputName || this.checkOutputName(inspector)) &&
+        (!this.exportName || this.checkExportName(inspector)) &&
+        (!this.outputValue || this.checkOutputValue(inspector));
```

Comment:
I feel like the `!this.outputName`, etc, are redundant. If they're not provided, the `check~` methods should probably just return `true` ("matched since no specific requirement").